modules/KalturaSupport/components/dualScreen/dualScreenCuePointsManager.js

```diff
@@ -0,0 +1,191 @@
+(function (mw, $) {
+    "use strict";
+    mw.dualScreen = mw.dualScreen || {};
+
+    mw.dualScreen.CuePointsManager = function(name, bindPostfix, player, onCuePointsReached) {
+
+        var $player = $(player);
+        var nextPendingCuePointIndex = null;
+
+        function log(context, message) {
+            mw.log(name + "." + context + ":" + message);
+        }
+
+        /**
+         * notify listener with cue points that need to be handled
+         * @param cuePoints
+         * @param eventContext
+         */
+
+        function triggerReachedCuePoints(cuePoints,eventContext)
+        {
+            if (onCuePointsReached) {
+                // create event args that contains functions to filter the cue points wisely
+                var eventArgs = $.extend({
+                    'cuePoints': cuePoints,
+                    filter: function (args) {
+                        var result = [];
+
+                        if (this.cuePoints) {
+                            result = $.grep(this.cuePoints, function (cuePoint) {
+                                var isValidTag = (!args.tag || (cuePoint.tags === args.tag));
+                                var isValidType = (!args.types || $.grep(args.types, function (cuePointType) {
+                                    return (!cuePointType.main || cuePointType.main === cuePoint.cuePointType) && (!cuePointType.sub || cuePointType.sub === cuePoint.subType);
+                                }).length > 0);
+
+                                return isValidTag && isValidType;
+                            });
+
+                            result.sort(function (a, b) {
+                                return args.sortDesc ? (b.startTime - a.startTime) : (a.startTime - b.startTime);
+                            });
+                        }
+
+                        return result;
+                    }
+                }, eventContext);
+
+                log('triggerReachedCuePoints()', 'notify listener with ' + cuePoints.length + ' cue points to handle');
+                onCuePointsReached(eventArgs);
+            }
+        }
+
+        function getCodeCuePoints() {
+            return player.kCuePoints.getCodeCuePoints();
+        }
+
+        function initialize() {
+            nextPendingCuePointIndex = getNextCuePointIndex(0);
+
+            $player.bind('onChangeMedia', function () {
+                nextPendingCuePointIndex = getNextCuePointIndex(0);
```

Comment:
@OrenMe does 'onChangeMedia' event is raised after the cuepoint service was updated with the new cuepoints?